daffodil-sapi/src/main/scala/org/apache/daffodil/sapi/Daffodil.scala

```diff
@@ -105,6 +106,10 @@ object ValidationMode extends Enumeration {
   val Off = Value(10)
   val Limited = Value(20)
   val Full = Value(30)
+
+  case class Custom(v: Validator) extends ValidationMode {
+    def id: Int = 100
+  }
```

Comment:
I would like to understand this part of the code better.  Why does Custom need to extend Value and add def id instead of simply assigning Value(100)?  Oh, I see Custom also adds a val v: Validator field and you use that v field further below in another file.  What about the id, why doesn't it need an `override` since it should be overriding Value's id though?

Reply:
Right, the custom mode provide a validator instance, and to do that it is a case class not a Val.

> What about the id, why doesn't it need an override since it should be overriding Value's id though?

Value is an abstract class and id is not defined.

Reply:
It can be a val though :wink: